code/datums/elements/food/food_trash.dm

```diff
@@ -0,0 +1,33 @@
+// If an item has the food_trash element it will drop an item when it is consumed.
+/datum/element/food_trash
+	element_flags = ELEMENT_BESPOKE
+	id_arg_index = 2
+	/// The type of trash that is spawned by this element
+	var/trash
+
+/datum/element/food_trash/Attach(datum/target, atom/trash)
+	. = ..()
+	if(!isatom(target))
+		return ELEMENT_INCOMPATIBLE
+	src.trash = trash
+	RegisterSignal(target, COMSIG_FOOD_CONSUMED, .proc/generate_trash)
+
+/datum/element/dunkable/Detach(datum/target)
+	. = ..()
+	UnregisterSignal(target, COMSIG_FOOD_CONSUMED)
+
+/datum/element/food_trash/proc/generate_trash(datum/source, mob/living/eater, mob/living/feeder)
+	SIGNAL_HANDLER
```

Comment:
you're sleeping in this proc, you run the risk of raising the ire of the dreaded moth maintainer
![image](https://user-images.githubusercontent.com/58055496/91287253-50f13980-e744-11ea-8c01-ad25d66b74f1.png)

Reply:
Something is calling/can call under the right circumstances do_mob 
![image](https://user-images.githubusercontent.com/58055496/91287483-a75e7800-e744-11ea-9d5b-8b321875a9cd.png)

Reply:
wtf ill have to look at the chain for that later

Reply:
what a dumb ass call chain

Reply:
Mood